blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
* https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManagerFactory.cxx#L213
run_recovery after createRMs
* https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManagerFactory.cxx#L172
resource manager is running recovery
* https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManager.cxx#L183
 calling txMangaer to send the recovery to the TM
1. the TM return 404 before merging the fix of JBTM-2356 and return false
2. the TM return 200 when merging the fix of JBTM-2356 and return true. So the rm delete the branch as it has been recovered but the HttpTxManager has still contains this branch.
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManager.cxx#L189
3. when HttpTxManager::request_handler() receive the COMMIT 
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/HttpTxManager.cxx#L349
the branch pointer should have been deleted and the SIGABORT happens.

Does it clear ?